js/viewer/annotation/base.js

```diff
@@ -27,69 +39,42 @@ function Base(attrs){
         }
         return false;
     }
-    /**
-     * This functions removes the style attributes, appends them to a string and returns them.
-     * @return {string}
-     */
-    this.getStyle = function() {
-        var styleString = "";
-        var styleAttributes = ['fill', "stroke", "stroke", "stroke-width"];
-
-        for (var i = 0; i <  styleAttributes.length; i++) {
-            if (this._attrs[styleAttributes[i]] && this._attrs[styleAttributes[i]].length > 0) {
-                styleString += styleAttributes[i] + ':' + this._attrs[styleAttributes[i]] + ';';
-                this._attrs[styleAttributes[i]] = null;
-            }
-        }
-
-        if (styleString[styleString.length - 1] == ';') {
-            styleString = styleString.slice(0, styleString.length - 1);
-        }
-
-        return styleString;
-    }
 
     this.exportToSVG = function(){
 
         // Check to see if there are necessary dimensions needed to construct the component. This makes sure that no empty components are added to the final SVG output file.
         if (!this.hasDimensions(this._attrs)) {
             return "";
         }
-
-        var tag = this._attrs["tag"];
+        var tag = this._tag;
         var rst = "<" + tag + " ";
         var attr;
-
-        // if(this.parent.id !== ""){
-        //     rst += 'id="' + this.parent.id + '" ';
-        // };
-
-        var style = this.getStyle();
-        if (style.length > 0) {
-            this._attrs["style"] = style;
-        }
+        var attributeList = {}
 
         for(attr in this._attrs){
-            // Ingoring style because that will be added at the end.
-            // Ignoreing vector-effect becasue this properpty is not needed in the output file
-            if (!this._attrs[attr] || this._attrs[attr] === null || attr === "style" || attr == "vector-effect"){
+            if (!this._attrs[attr] || this._attrs[attr] === null || attr === "graph-id"){
                 continue;
             }
 
-            switch(attr){
-                case "tag":
-                case "graph-id":
-                    break;
-                default:
-                    rst += (attr + '="' + this._attrs[attr] + '" ');
-                    break;  
+            if (!(this._addedProperties[attr])) {
+                attributeList[attr] = this._attrs[attr];
             }
         }
 
-        // The style is added at the end
-        if (this._attrs["style"].length > 0) {
-            rst += ('style="' + this._attrs["style"] + '" ');
+        for (attr in this._ignoredAttributes) {
+            attributeList[attrs] = this._ignoredAttributes[attr];
         }
+
+        // sort the attributes to make the output deterministic
+        var sortedAttributeList = {};
+        Object.keys(attributeList).sort().forEach(function (key) {
+            sortedAttributeList[key] = attributeList[key];
+        });
+
+        for(attr in sortedAttributeList){
```

Comment:
Iterating over objects doesn't necessarily preserve the order. You have the keys sorted up there, so add to output directly up there. You don't need to create another object, so

```javascript 
Object.keys(attributeList).sort().forEach(function (key) {
  rst += (key + '="' + attributeList[key] + '" ');
});
```

Reply:
Done